emukit/core/optimization/gradient_acquisition_optimizer.py

```diff
@@ -52,10 +54,10 @@ def f_df(x):
             f_df = None
 
         optimizer = self._get_optimizer(context_manager)
-        anchor_points_generator = ObjectiveAnchorPointsGenerator(self.space, acquisition)
+        anchor_points_generator = ObjectiveAnchorPointsGenerator(self.space, acquisition, num_samples=self.num_samples)
 
         # Select the anchor points (with context)
-        anchor_points = anchor_points_generator.get(num_anchor=1, context_manager=context_manager)
+        anchor_points = anchor_points_generator.get(num_anchor=self.num_anchor, context_manager=context_manager)
```

Comment:
That seems to change the default/previous behavior. Can `num_anchor` be 1 by default? Actually I suspect this change is not that severe to make a massive difference. What are your thoughts?

Reply:
That's a good point. The issue here is that using one anchor point is insufficient to optimize the acquisition properly, so it makes sense to be explicit about it and ask the user to select it (or use a larger number by default perhaps?)

Reply:
I agree, it's almost always better to have `num_anchors` > 1. Our wrapper has its own control for this anyway, so it's not important for us, but perhaps we should change the `GradientAcquisitionOptimizer.init()` to have `num_anchors=1` by default instead of `num_anchors=5`, for consistency with the existing behaviour?

Reply:
Changed.